kuka_rox_hardware_interface/src/robot_manager_node.cpp

```diff
@@ -124,9 +156,52 @@ RobotManagerNode::on_cleanup(const rclcpp_lifecycle::State &)
   return SUCCESS;
 }
 
+void RobotManagerNode::ObserveControl()
+{
+  #ifdef NON_MOCK_SETUP
+  while (!terminate_) {
+    context_ = std::make_unique<::grpc::ClientContext>();
+    kuka::ecs::v1::ObserveControlStateRequest observe_request;
+    std::unique_ptr<grpc::ClientReader<kuka::ecs::v1::CommandState>> reader(
+      stub_->ObserveControlState(context_.get(), observe_request));
+
+    kuka::ecs::v1::CommandState response;
+
+    if (reader->Read(&response)) {
+      switch (static_cast<int>(response.event())) {
+        case kuka::ecs::v1::CommandEvent::STOPPED:
+        case kuka::ecs::v1::CommandEvent::ERROR:
+          RCLCPP_INFO(get_logger(), "External control stopped");
+          terminate_ = true;
+          if (this->get_current_state().label() == "active") {
```

Comment:
Can't we work with this->get_current_state().id() instead? It's a bit better than hardcoding strings imo.

Reply:
fully agree :)

Reply:
is there an enum for the id? I think I could not find one and therefore used the string